### PR TITLE
Fixing issue with hashtag symbols in url. Issue #556

### PR DESCRIPTION
**Pull request by laraconda**
## Summary of the changes in this pull request

- Editor now avoids marking cpp directives as hashtags
- Editor now correctly colors urls containing '#' symbols

* TODO

## Pull request checklist

- [x] I have added an entry in `CHANGELOG.md` including my name and issue and/or pull request number.
- [ ] If applicable: I have removed the corresponding entry in `TODO.md`.

### CODE REVIEW

**Comment by laraconda** [comment]
I unfortunately discovered a new issue regarding pattern recognition. In preview mode urls and hashtags cause problems. This is related to the processing made by t2t. I'll open a new issue to discuss details there.

File: rednotebook/files/t2t.lang [hunk 1]
```diff
@@ -243,15 +243,26 @@
     </context>
 
     <context id="bare-url" style-ref="url">
-      <match>\b((https?|ftp|news|telnet|gopher|wais)://|(www[23]?|ftp)\.)[A-Za-z0-9%._/~:,=$@+-]+\b/*(\?[A-Za-z0-9/%&amp;=+:;.,$@*_-]+)?(#[A-Za-z0-9%._-]*)?</match>
+        <match>\b((https?|ftp|news|telnet|gopher|wais)://|(www[23]?|ftp)\.)[A-Za-z0-9%._/~:,=$@+-]+\b/*(\?[A-Za-z0-9/%&amp;=+:;.,$@*_-]+)?(#[/A-Za-z0-9%._-]*)?</match>
```
Comment (jendrikseipp) [inline rednotebook/files/t2t.lang:246]:
```suggestion
      <match>\b((https?|ftp|news|telnet|gopher|wais)://|(www[23]?|ftp)\.)[A-Za-z0-9%._/~:,=$@+-]+\b/*(\?[A-Za-z0-9/%&amp;=+:;.,$@*_-]+)?(#[/A-Za-z0-9%._-]*)?</match>
```

**Review by jendrikseipp — APPROVED** [review]
Looks good to me, thanks! Please also add a short changelog entry.

**Comment by jendrikseipp** [comment]
Thanks!

**Pull request merged**